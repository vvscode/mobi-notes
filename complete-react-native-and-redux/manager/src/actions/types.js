export const EMAIL_CHANGED = 'email_changed';
export const PASSWORD_CHANGED = 'password_changed';

export const LOGIN_USER_START = 'login_user_start';
export const LOGIN_USER_SUCCESS = 'login_user_success';
export const LOGIN_USER_FAIL = 'login_user_fail';

export const EMPLOYEE_FIELD_UPDATE = 'employee_field_update';
export const EMPLOYEE_CREATE_START = 'employee_create_start';
export const EMPLOYEE_CREATE_SUCCESS = 'employee_create_success';
export const EMPLOYEE_CREATE_FAIL = 'employee_create_fail';

export const EMPLOYEE_LIST_LOADING_START = 'employee_list_loading_start';
export const EMPLOYEE_LIST_LOADING_SUCCESS = 'employee_list_loading_success';
export const EMPLOYEE_LIST_LOADING_FAIL = 'employee_list_loading_fail';