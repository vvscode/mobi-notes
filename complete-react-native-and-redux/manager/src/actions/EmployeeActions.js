import firebase from 'firebase';
import { Actions } from 'react-native-router-flux';

import * as types from './types';

export const employeeFieldUpdate = ({ prop, value }) => ({
  type: types.EMPLOYEE_UPDATE,
  payload: { prop, value },
});

export const employeeCreate = ({ name, phone, shift }) => (dispatch) => {
  dispatch({ type: types.EMPLOYEE_CREATE_START });
  const { currentUser: { uid } } = firebase.auth();
  firebase.database()
    .ref(`/users/${uid}/employees`)
    .push({name, phone, shift })
    .then((employee) => dispatch({
      type: types.EMPLOYEE_CREATE_SUCCESS,
      pyaload: employee
    }))
    .then(() => Actions.employeeList())
    .catch((err) => dispatch({
      type: types.EMPLOYEE_CREATE_FAIL,
      payload: err
    }));
};

export const employeeFetch = () => (dispatch) => {
  dispatch({ type: types.EMPLOYEE_LIST_LOADING_START });
  const { currentUser: { uid } } = firebase.auth();
  const employeesRef = firebase.database().ref(`/users/${uid}/employees`);
  employeesRef.on('value', (snapshot) => dispatch({
    type: types.EMPLOYEE_LIST_LOADING_SUCCESS,
    payload: snapshot.val()
  }));
}
