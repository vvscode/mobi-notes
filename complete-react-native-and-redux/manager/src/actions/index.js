import firebase from 'firebase';
import * as types from './types';

export const emailChanged = (text) => ({type: types.EMAIL_CHANGED, payload: text});

export const passwordChanged = (text) => ({type: types.PASSWORD_CHANGED, payload: text});

export const loginFormSubmit = ({email, password}) => (dispatch) => {
  dispatch({type: types.LOGIN_USER_START});
  let firebaseAuth = firebase.auth();
  Promise
    .resolve()
    .then(() => firebaseAuth.signInWithEmailAndPassword(email, password))
    .catch(() => firebaseAuth.createUserWithEmailAndPassword(email, password))
    .then((user) => dispatch({type: types.LOGIN_USER_SUCCESS, payload: user}))
    .catch(({message}) => dispatch({type: types.LOGIN_USER_FAIL, payload: message}));
};