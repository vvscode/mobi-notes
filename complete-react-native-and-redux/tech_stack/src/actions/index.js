export const selectLibrary = (libararyId) => ({
  type: 'select_library',
  payload: libararyId
});