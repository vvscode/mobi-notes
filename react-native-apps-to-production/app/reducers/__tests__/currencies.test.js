import reducer from '../currencies';
import { getInitialConversion } from '../../actions/currencies';

it('sets inital state', () => {
  expect(reducer(undefined, {})).toMatchSnapshot();
});

it('sets nested data on initial fetch', () => {
  expect(reducer(undefined, getInitialConversion())).toMatchSnapshot();
});