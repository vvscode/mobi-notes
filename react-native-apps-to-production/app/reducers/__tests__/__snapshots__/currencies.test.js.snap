// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets inital state 1`] = `
Object {
  "amount": 100,
  "baseCurrency": "USD",
  "conversions": Object {},
  "error": null,
  "quoteCurrency": "GBP",
}
`;

exports[`sets nested data on initial fetch 1`] = `
Object {
  "amount": 100,
  "baseCurrency": "USD",
  "conversions": Object {
    "USD": Object {
      "date": "",
      "isFetching": true,
      "rates": Object {},
    },
  },
  "error": null,
  "quoteCurrency": "GBP",
}
`;
